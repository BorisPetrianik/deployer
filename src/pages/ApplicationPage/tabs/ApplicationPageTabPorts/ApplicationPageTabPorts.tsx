import React from 'react'
<<<<<<< Updated upstream
import { makeStyles } from '@material-ui/styles'
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
  Button,
  TextField,
  Grid,
  // Checkbox,
} from '@material-ui/core'
import { IPortMapping } from '../../../../types/Application'
=======
import { IPorts } from '../../../../types/Application'
>>>>>>> Stashed changes

interface IApplicationPageTabPortsProps {
  ports: IPorts
}
const useStyles = makeStyles({
  tableContainer: {
    marginLeft: '30%',
    width: '650px',
  },
  saveBtn: {
    marginTop: '5px',
    padding: '10px 60px',
    backgroundColor: '#dc70e6',
  },
})

const preparePorts = (ports: IPortMapping[]) =>
  ports.reduce((accum, value) => ({ ...accum, [value.port]: value.value }), {})

export const ApplicationPageTabPorts = ({ ports }: IApplicationPageTabPortsProps) => {
  const classes = useStyles()
  const [port, setPorts] = React.useState<{ [key: string]: string }>(preparePorts(ports))
  const handleChangePorts = (event: any) => {
    const { name, value } = event.target
    setPorts((prevState) => ({ ...prevState, [name]: value }))
  }
  return (
    <Grid className={classes.tableContainer} container direction='column' justify='center' alignItems='center'>
      <h3>ApplicationPageTabPorts</h3>
      <TableContainer component={Paper}>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Port</TableCell>
              <TableCell align='right'>Value</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {Object.keys(port).map((por) => (
              <TableRow key={por}>
                <TableCell>
                  <TextField disabled id='standard-required' variant='filled' value={por} />
                </TableCell>
                <TableCell align='right'>
                  <TextField id='standard' name={por} variant='filled' value={port[por]} onChange={handleChangePorts} />
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
      <Button className={classes.saveBtn} variant='contained'>
        Save
      </Button>
    </Grid>
  )
}
