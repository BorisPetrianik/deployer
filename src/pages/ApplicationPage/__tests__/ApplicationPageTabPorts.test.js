import React from 'react'
import renderer from 'react-test-renderer'
import { ApplicationPageTabPorts } from '../tabs/ApplicationPageTabPorts/ApplicationPageTabPorts'

it('renders correctly', () => {
    const tree = renderer
        .create(<ApplicationPageTabPorts ports={['88:80']}/>)
        .toJSON()
    
    expect(tree).toMatchSnapshot()
})
