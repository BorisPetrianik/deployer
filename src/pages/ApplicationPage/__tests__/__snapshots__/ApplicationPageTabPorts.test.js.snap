// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    ApplicationPageTabPorts
  </h3>
  ["88:80"]
</div>
`;
